### ISSUE TITLE
Decouple dmf_control_board plugin (Trac #89)

### ISSUE THREAD

**Issue by cfobel** [wheeler-microfluidics/microdrop#93]
Rather than being a special case, the dmf_control_board should be decoupled from the microdrop application to behave like any other plugin. This would give us the option to use different versions of the dmf_control_board which could correspond, for example, to different hardware devices.

Perhaps it is worth writing a plugin interface for communicating with dmf_control_board objects?

Migrated from http://microfluidics.utoronto.ca/microdrop/ticket/89
```json
{
    "status": "closed", 
    "changetime": "2014-04-17T19:39:01", 
    "description": "Rather than being a special case, the dmf_control_board should be decoupled from the microdrop application to behave like any other plugin. This would give us the option to use different versions of the dmf_control_board which could correspond, for example, to different hardware devices.\n\nPerhaps it is worth writing a plugin interface for communicating with dmf_control_board objects?", 
    "reporter": "ryan", 
    "cc": "", 
    "resolution": "fixed", 
    "_ts": "1397763541728826", 
    "component": "microdrop", 
    "summary": "Decouple dmf_control_board plugin", 
    "priority": "major", 
    "keywords": "", 
    "version": "0.1", 
    "time": "2012-04-02T16:32:50", 
    "milestone": "Microdrop 1.0", 
    "owner": "cfobel", 
    "type": "task"
}
```

**Comment by cfobel** [comment]
Trac update at 20120403T15:16:31: 
* ryan changed owner from "ryan" to "cfobel"
* ryan changed status from "new" to "assigned"

**Comment by cfobel** [comment]
Trac update at 20120407T05:21:07: christian@fobel.net commented:

> commit 65de479d0730698c02d58d2aa487aa9bd8cb0c35
> Author: Christian Fobel <christian@fobel.net>
> Date:   Sat Apr 7 01:00:30 2012 -0400
> 
>     [SUBMODULE] Update protocol grid on enable/disable
>     
>     Protocol grid view should be updated when the dmf_control_board plugin is
>     enabled or disabled.
>     
>     See #93.

**Comment by cfobel** [comment]
Trac update at 20120407T05:21:11: 
* christian@fobel.net commented:

> commit 77cbab42db16aa92e6439c83b21a368e03e85e8f
> Author: Christian Fobel <christian@fobel.net>
> Date:   Wed Apr 4 19:39:52 2012 -0400
> 
>     [SUBMODULE] Decouple dmf_control_board plugin
>     
>     Fixes #93.
>     Convert DmfControlBoardPlugin to a Plugin.
>     Eliminate app.control_board and any references to it.
>     Skip updates that involve a device or experiment log if the corresponding
>     app attribute is not ready.
>     
>     Note that the experiment log plugin and protocol controller refer to the
>     dmf_control_board plugin.  However, they should now run fine without the
>     dmf_control_board installed.


* christian@fobel.net changed resolution from "" to "fixed"
* christian@fobel.net changed status from "assigned" to "closed"

**Comment by cfobel** [comment]
Trac update at 20140417T19:39:01: ryan changed component from "" to "microdrop"
